internal/configs/config.go

```diff
@@ -950,13 +951,42 @@ func (c *Config) TransformForTest(run *TestRun, file *TestFile) (func(), hcl.Dia
 		// Otherwise, let's copy over and overwrite all providers specified by
 		// the test file itself.
 		for key, provider := range file.Providers {
-			next[key] = provider
+			next[key] = &Provider{
+				Name:       provider.Name,
+				NameRange:  provider.NameRange,
+				Alias:      provider.Alias,
+				AliasRange: provider.AliasRange,
+				Version:    provider.Version,
+				Config:     &TestProviderConfig{Body: provider.Config, Value: cty.NilVal},
```

Comment:
I'm a little concerned we may be mixing up real nil values and cty.NilVal here.

Is there a reason the Value field isn't a *cty.Value so we can just pass nil instead of cty.NilVal?

I would rather check if the value is unset, rather than cty.NilVal